lib/puma/reactor.rb

```diff
@@ -302,6 +303,7 @@ def clear_monitor(mon)
     def run
       run_internal
     ensure
+      @selector.close
```

Comment:
Same here, was this in response to a specific issue?

Reply:
nio4r doc for the method: 'Close the selector and free system resources'

I don't recall whether it affected leaky fd's or not.  It should be done somewhere...